lean4/src/putnam_1989_b6.lean

```diff
@@ -1,23 +1,19 @@
 import Mathlib
 
-open Nat Filter Topology Set
+open Nat Filter Topology Set ProbabilityTheory
 
 -- Note: uses (ℝ → ℝ) instead of (Set.Icc 0 1 → ℝ)
 /--
 Let $(x_1,x_2,\dots,x_n)$ be a point chosen at random from the $n$-dimensional region defined by $0<x_1<x_2<\dots<x_n<1$. Let $f$ be a continuous function on $[0,1]$ with $f(1)=0$. Set $x_0=0$ and $x_{n+1}=1$. Show that the expected value of the Riemann sum $\sum_{i=0}^n (x_{i+1}-x_i)f(x_{i+1})$ is $\int_0^1 f(t)P(t)\,dt$, where $P$ is a polynomial of degree $n$, independent of $f$, with $0 \leq P(t) \leq 1$ for $0 \leq t \leq 1$.
 -/
 theorem putnam_1989_b6
-(n : ℕ)
-(Sx : Set (Fin n → ℝ))
-(fprop : (ℝ → ℝ) → Prop)
-(xext : (Fin n → ℝ) → (ℕ → ℝ))
-(fxsum : (ℝ → ℝ) → (Fin n → ℝ) → ℝ)
-(fEV : (ℝ → ℝ) → ℝ)
-(hSx : Sx = {x : Fin n → ℝ | 0 < x ∧ StrictMono x ∧ x < 1})
-(hfprop : fprop = (fun f : ℝ → ℝ => ContinuousOn f (Set.Icc 0 1) ∧ f 1 = 0))
-(hfxsum : fxsum = (fun (f : ℝ → ℝ) (x : Fin n → ℝ) => ∑ i in Finset.Icc 0 n, ((xext x) (i + 1) - (xext x) i) * f ((xext x) (i + 1))))
-(hfEV : fEV = (fun f : ℝ → ℝ => (∫ x in Sx, fxsum f x) / (∫ x in Sx, 1)))
-(npos : n ≥ 1)
-(hxext : ∀ x : Fin n → ℝ, (xext x) 0 = 0 ∧ (xext x) (n + 1) = 1 ∧ (∀ i : Fin n, (xext x) (i + 1) = x i))
-: ∃ P : Polynomial ℝ, P.degree = n ∧ (∀ t ∈ Set.Icc 0 1, 0 ≤ P.eval t ∧ P.eval t ≤ 1) ∧ (∀ f : ℝ → ℝ, fprop f → fEV f = (∫ t in Set.Ioo 0 1, f t * P.eval t)) :=
-sorry
+  (n : ℕ) (hn : 0 < n)
+  (S : (ℝ → ℝ) → (Fin (n + 2) → Icc (0 : ℝ) 1) → ℝ)
+  (hS : ∀ f x, S f x = if StrictMono x ∧ x 0 = 0 ∧ x (-1) = 1 then ∑ i in Icc 0 n, (x (i + 1) - x i) * f (x (i + 1)) else 0)
+  : ∃ P : Polynomial ℝ,
+    P.degree = n ∧
+    (∀ t ∈ Icc 0 1, P.eval t ∈ Icc 0 1) ∧
+    (∀ f : ℝ → ℝ,
+      f 1 = 0 ∧ ContinuousOn f (Icc 0 1) →
+      𝔼[(↑) ∘ (S f)] = ∫ t in (0)..1, (f t) * (P.eval t)) :=
```

Comment:
I think this would be better as
```suggestion
      𝔼[fun x => S f x] = ∫ t in (0)..1, f t * P.eval t) :=
```
which makes it clear where the `x` in the question has got to